plugins/woocommerce/src/Internal/Admin/Logging/PageController.php

```diff
@@ -161,13 +164,129 @@ private function render_file_list_page( array $params = array() ): void {
 		<?php
 	}
 
+	/**
+	 * Render the single file view.
+	 *
+	 * @param array $params Args for rendering the view.
+	 *
+	 * @return void
+	 */
+	private function render_single_file_page( array $params ): void {
+		$file = $this->file_controller->get_file_by_id( $params['file_id'] );
+
+		if ( is_wp_error( $file ) ) {
+			?>
+			<div class="notice notice-error notice-inline">
+				<?php echo wp_kses_post( wpautop( $file->get_error_message() ) ); ?>
+				<?php
+				printf(
+					'<p><a href="%1$s">%2$s</a></p>',
+					esc_url( $this->get_logs_tab_url() ),
+					esc_html__( 'Return to the file list.', 'woocommerce' )
+				);
+				?>
+			</div>
+			<?php
+
+			return;
+		}
+
+		$rotations         = $this->file_controller->get_file_rotations( $file->get_file_id() );
+		$rotation_url_base = add_query_arg( 'view', 'single_file', $this->get_logs_tab_url() );
+
+		$delete_url = add_query_arg(
+			array(
+				'action'  => 'delete',
+				'file_id' => array( $file->get_file_id() ),
+			),
+			wp_nonce_url( $this->get_logs_tab_url(), 'bulk-log-files' )
+		);
+
+		$stream      = $file->get_stream();
+		$line_number = 1;
+
+		$delete_confirmation_js = sprintf(
+			"return window.confirm( '%s' )",
+			esc_js( __( 'Delete this log file permanently?', 'woocommerce' ) )
+		);
+
+		?>
+		<header id="logs-header" class="wc-logs-header">
+			<h2>
+				<?php
+				printf(
+				// translators: %s is the name of a log file.
```

Comment:
```suggestion
					// translators: %s is the name of a log file.
```

I'm not sure why this is so common, my own IDE seems to often do the same thing 🤔